expression/builtin_compare.go

```diff
@@ -1090,17 +1090,17 @@ func RefineComparedConstant(ctx sessionctx.Context, isUnsigned bool, con *Consta
 		}
 	case opcode.NullEQ, opcode.EQ:
 		switch con.RetType.EvalType() {
-		// An integer value equals or null-equals to a float value contains
-		// non-zero decimal digits is definite false.
-		// e.g:
-		//   1. "integer  =  1.1" is definite false.
-		//   2. "integer <=> 1.1" is definite false.
+		// An integer value equal or not equal to a float value which contains
```

Comment:
s/ not equal/ NULL-safe equal
refer: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#operator_equal-to